contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,959 @@
+        return _asset == WETH;
+    }
+
+    /**
+     * @dev Internal method to respond to the addition of new asset / pTokens
+            We need to give the Aerodrome pool approval to transfer the
+            asset.
+     */
+    function _abstractSetPToken(address, address) internal override {
+        // the deployer shall call safeApproveAllTokens() to set necessary approvals
+        revert("Unsupported method");
+    }
+
+    /**
+     * @dev Approve the spending of all assets
+     */
+    function safeApproveAllTokens()
+        external
+        override
+        onlyGovernor
+        nonReentrant
+    {
+        // to add liquidity to the clPool
+        IERC20(WETH).safeApprove(address(positionManager), type(uint256).max);
+        IERC20(OETHb).safeApprove(address(positionManager), type(uint256).max);
+        // to be able to rebalance using the swapRouter
+        IERC20(WETH).safeApprove(address(swapRouter), type(uint256).max);
+        IERC20(OETHb).safeApprove(address(swapRouter), type(uint256).max);
+    }
+
+    /***************************************
+            Balances and Fees
+    ****************************************/
+
+    /**
+     * @dev Get the total asset value held in the platform
+     * @param _asset      Address of the asset
+     * @return balance    Total value of the asset in the platform
+     */
+    function checkBalance(address _asset)
+        external
+        view
+        override
+        returns (uint256)
+    {
+        require(_asset == WETH, "Only WETH supported");
+
+        // we could in theory deposit to the strategy and forget to call rebalance in the same
+        // governance transaction batch. In that case the WETH that is on the strategy contract
+        // also needs to be accounted for.
+        uint256 _wethBalance = IERC20(WETH).balanceOf(address(this));
+        // just paranoia check, in case there is OETHb in the strategy that for some reason hasn't
+        // been burned yet.
+        uint256 _oethbBalance = IERC20(OETHb).balanceOf(address(this));
+        return underlyingAssets + _wethBalance + _oethbBalance;
+    }
+
+    /**
+     * @dev Returns the balance of both tokens in a given position (excluding fees)
+     * @return _amountWeth Amount of WETH in position
+     * @return _amountOethb Amount of OETHb in position
+     */
+    function getPositionPrincipal()
+        public
+        view
+        returns (uint256 _amountWeth, uint256 _amountOethb)
+    {
+        if (tokenId == 0) {
+            return (0, 0);
+        }
+
+        uint160 _sqrtRatioX96 = getPoolX96Price();
+        (_amountWeth, _amountOethb) = helper.principal(
+            positionManager,
+            tokenId,
+            _sqrtRatioX96
+        );
+    }
+
+    /**
+     * @notice Returns the current pool price in X96 format
+     * @return _sqrtRatioX96 Pool price
+     */
+    function getPoolX96Price() public view returns (uint160 _sqrtRatioX96) {
```

Comment:
This probably doesn't need to be public, since this can be just be asked to the pool?

Reply:
agreed that it doesn't need to be, though I think it is a nice convenience function, since querying from the pool returns the whole `slot0` struct data and that is more cumbersome. 
I would keep it as convenience, though I don't feel strongly about it.

Reply:
Keep is fine then.